net/dnscache/dnscache.go

```diff
@@ -31,19 +31,19 @@ import (
 var zaddr netip.Addr
 
 var single = &Resolver{
-	Forward: &net.Resolver{PreferGo: preferGoResolver()},
+	Forward: &net.Resolver{PreferGo: preferGoResolver(runtime.GOOS)},
```

Comment:
We can't change this one. Setting PreferGoResolver to true on Windows isn't good unless you're also overriding the dialer which this isn't.

Reply:
The behaviour here shouldn't have changed. Was this comment intended for a different line?

Reply:
The bug you referenced (#5161) is a big project. It's not as simple as flipping a bool. This change doesn't look correct. Did you even test it?